bin/ruby-build

```diff
@@ -1022,20 +1023,28 @@ locate_llvm() {
 }
 
 needs_yaml() {
-  [[ "$RUBY_CONFIGURE_OPTS" != *--with-libyaml-dir=* ]] &&
-  ! use_homebrew_yaml
+  [ -z "$(homebrew_yaml_dir)" ]
 }
 
 use_homebrew_yaml() {
-  local libdir="$(brew --prefix libyaml 2>/dev/null || true)"
-  if [ -d "$libdir" ]; then
+  local libdir="$(homebrew_yaml_dir)"
+  if [ ! -z "$libdir" ]; then
     echo "ruby-build: using libyaml from homebrew"
     package_option ruby configure --with-libyaml-dir="$libdir"
   else
     return 1
   fi
 }
 
+homebrew_yaml_dir() {
+  if [[ "$RUBY_CONFIGURE_OPTS" != *--with-libyaml-dir=* ]]; then
+    local libdir="$(brew --prefix libyaml 2>/dev/null || true)"
+    if [[ -d "$libdir" ]]; then
+      echo "$libdir"
+    fi
+  fi
+}
```

Comment:
I've moved common parts to a separate function so that it's possible to check whether yaml is needed without modifying package options